src/Horus/Global.hs

```diff
@@ -347,8 +352,22 @@ solveContract = do
     )
       infos
  where
-  isStandardSource :: Set ScopedFunction -> ScopedFunction -> Bool
-  isStandardSource inlinables f = f `notElem` inlinables && not (isWrapper f)
+  standardSources :: Set ScopedFunction -> [LabeledInst] -> GlobalL (Set ScopedFunction)
+  standardSources inlinableFs rows =
+    getProgram >>= \prog ->
+      let functionsWithBodies = functionsOf rows prog
+          functions = Map.keys functionsWithBodies
+          (cg, vToLbl, lblToV) = graphOfCG . callgraph . Map.mapKeys sf_pc $ functionsWithBodies
+          (wrapperFunctions, nonwrapperFunctions) = partition isWrapper functions
+          reachableLsFromWrappers =
+            Set.fromList
+              . concatMap (concatMap ((^. _3) . vToLbl) . reachable cg . fromJust . lblToV . sf_pc)
```

Comment:
How painful would it be to remove the `fromJust` call?

Reply:
Very, because of dependence on `graphFromEdges`, which turns a custom Vertex representation to its internal `Int`-based one and gives you what is essentially a `Key -> Maybe Int` map. The `fromJust` is safe by virtue of the stuff being queried on a partition of functions from which the graph itself had been made prior.

Reply:
Okay, fair enough!